src/kernl/utils/debugger.py

```diff
@@ -159,6 +161,7 @@ def get_ptr(self, tensor: torch.Tensor) -> int:
         :param tensor: input tensor
         :return: pointer as an integer
         """
+        self._add_input(tensor)
```

Comment:
what do you mean ? You can just check if the tensor is in the map, no ?

Reply:
for example if we have a zeros tensor, and we want to add another zeros tensor we can't just check and say it already exists so we won't add it (that's what  i mean).

Reply:
i think you need to check existance by checking if there is a tensor with same reference, not by comparing value.

Reply:
@gaetansnl I don't know if we really can do this, because as you can see we are passing only the tensor and we are creating the reference when calling get_ptr. We don't have a reference to compare it and to test if it really exists or not. the reference is defined by the boundary and we are incrementing it with each add.